app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_counter
```

Comment:
- [optional] After using the Rails Active Callbacks(`after_save`), to call the `update_counter` method then you need to make the `update_counter` method private. Please do the same for `comments` and `likes.`